### PR TITLE
Implementation of the doorhanger UI

### LINKED ISSUE #14: New UI treatment (doorhanger) needs to be implemented
See [designer mockup](https://mozilla.invisionapp.com/share/WECDBAXD6#/screens/241301930_CopyURL).

### LINKED ISSUE #16: Right click in doorhanger causes context menu TypeError
`TypeError: gContextMenuContentData is null[Learn More]`
`TypeError: gContextMenu is null[Learn More]`

### LINKED ISSUE #17: Background color of the popup panel HTML is not correct
Plan to fix:
1. Remove padding from the panel around the embedded HTML. See `social-share-panel` for example. Inject this CSS just like `share_button.css`.
2. Apply [OS-specific panel gradient](https://dxr.mozilla.org/mozilla-central/search?q=%22--arrowpanel-background%3A%22&redirect=false).

### LINKED ISSUE #23: Doorhanger test is not working reliably
See:
- [build 113](https://circleci.com/gh/marcrowo/share-button-study/113)
- [build 112](https://circleci.com/gh/marcrowo/share-button-study/112)

### PR DESCRIPTION
**Pull request by marcrowo**
- Fixes #14.
- Fixes #16.
- Fixes #17.
- Fixes #23: doorhanger tests are now working! Needed to add a window manager for panels to show up. See issue for more details.

### CODE REVIEW

File: extension/bootstrap.js [hunk 1]
```diff
@@ -3,13 +3,57 @@ Cu.import("resource://gre/modules/Services.jsm");
...
+function doorhangerTreatment(browserWindow, shareButton) {
+  let panel = browserWindow.window.document.getElementById("share-button-panel");
+  if (panel === null) { // create the panel
+    panel = browserWindow.window.document.createElement("panel");
+    const props = {
+      id: "share-button-panel",
+      type: "arrow",
+      noautofocus: true,
+      level: "parent",
+      style: "width:400px; height:96px;",
```
Comment (mythmon) [inline extension/bootstrap.js]: Should this be a part of `panel.css`?

File: manual_test.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+/* eslint-env node */
+
+const utils = require("./test/utils");
+// const webdriver = require("selenium-webdriver");
```
Comment (mythmon) [inline manual_test.js]: You probably don't want this here anymore.

File: manual_test.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+/* eslint-env node */
```
Comment (mythmon) [inline manual_test.js:1]: You should probably add a comment explaining what this file is for.

File: test/button_test.js [hunk 4]
```diff
@@ -67,23 +74,37 @@ describe("Add-on Functional Tests", function() {
...
   it("should no longer trigger animation once uninstalled", async() => {
     await utils.uninstallAddon(driver, addonId);
     await utils.copyUrlBar(driver);
     const { hasClass, hasColor } = await utils.testAnimation(driver);
     assert(!hasClass && !hasColor);
   });
+
+  it("should no longer trigger popup once uninstalled", async() => {
+    await utils.copyUrlBar(driver);
+    await utils.copyUrlBar(driver);
```
Comment (mythmon) [inline test/button_test.js]: Why do this twice?

**Review by mythmon — CHANGES_REQUESTED** [review]
I have a few suggestions, but overall this looks good. Nice work.

File: test/button_test.js (hunk 4 above)
Comment (marcrowo) [reply test/button_test.js]: Oops, that should not be there.

File: manual_test.js (hunk 2 above)
Comment (marcrowo) [reply manual_test.js]: ~~Correct, this is not needed as I moved all of the `webdriver` references to `utils.js`. I think that this is probably better to keep the `button_test.js` as clear and clean as possible.~~

Oops, I was looking at `button_test.js`. In both files this require comment is not necessary.

**Review by mythmon — APPROVED** [review]

**Comment by marcrowo** [comment]
Having some issues with testing triggering the popup, so I am not merging this yet.

**Comment by marcrowo** [comment]
Fixes #23: doorhanger tests are now working! Needed to add a window manager for panels to show up. See issue for more details.

**Pull request merged**